tla-assignments/src/main/scala/at/forsyte/apalache/tla/assignments/SymbTransGenerator.scala

```diff
@@ -242,42 +234,6 @@ class SymbTransGenerator( tracker : TransformationTracker ) {
       strategy.filter( selection.contains ) // strategy is already ordered
     }
 
-    /**
-      * Constructs a ranking function which assigns
-      * to each `tlaEx` the following value:
-      *   - Some( min S ), if the set filtered = { e.ID | e is a subexpression of tlaEx and s.contains( e.ID ) } of
-      *     assignment subexpression IDs is nonempty and the set S = { j | orderedAssignments[j] \in filtered }
-      *     of positions those assignments hold w.r.t. orderedAssignments is nonempty
-      *   - None otherwise
-      *
-      * Example:
-      *   s = { 1, 3, 42},
-      *   orderedAssignments = [ 3, 42, 1 ]
-      *   tlaEx = /\ x' = 1 (uid 1)
-      *           /\ x' = 2 (uid 2)
-      *           /\ y' = 3 (uid 3)
-      *
-      *   Then, filtered = { 1, 3 }, S = { 0, 2 } and the ranking function for tlaEx returns Some( 0 )
-      *   (i.e. in reordering, tlaEx has the same rank as its lowest-ranking assingment subexpression `y' = 3`)
-      */
-    private def mkRankingFun( s: Set[UID],
-                              selections : SelMapType,
-                              orderedAssignments: StrategyType
-                            ) : TlaEx => Option[Int] = { tlaEx =>
-      // the union of all sets in selections( tlaEx.ID ) lists all the assignments in the subexpressions of TlaEx
-      val assignmentsBelow = selections.getOrElse( tlaEx.ID, Set.empty[Set[UID]] ).foldLeft( Set.empty[UID] ){ _ ++ _ }
-      // We filter them via the current selection (to ignore assignments on a branch we're not currently on)
-      val filtered = assignmentsBelow.intersect( s )
-      // Of all the assignment candidates, the one with the lowest position in `orderedAssignments` determines the rank
-      filtered.foldLeft( None : Option[Int] ) {
-        case (rankCand, uid) =>
-          val uidRank = orderedAssignments.find( _ == uid ) map {
-            orderedAssignments.indexOf
-          }
-          Reordering.IntOptOrdering.min( rankCand, uidRank )
-      }
-    }
-
```

Comment:
Does it mean that we have completely get rid of the SMT version? I would keep it on a side and mark it as deprecated. It is still useful to have that code. We might want to use it in the future, perhaps, when we have a different SMT encoding in the model checker.

Reply:
If we decide to keep the code as deprecated, then how about wrapping this deprecated solver in a feature flag (that can be passed via CLI)? Otherwise, if the code is not going to be called at all by the application I really don't think we should we keep it. We can always recover it from the git history (i.e., we already have files with this code in it), but leaving dead code in the code base is liable to lead to errors down the line.

Reply:
If the new version works for us, let's drop the support for the old implementation. But in order to find easily, we may want to introduce a clear tag that shows where the divergence has begun.

Reply:
Just to comment on this a bit, the code removed here relates only to reordering, which is now unnecessary. I've opened a separate issue for removing SMT-related code in the future (#367)